### DISCUSSION TITLE
How to create a condition for a loop node?

### DISCUSSION THREAD

**Discussion by martinakavanova** [jspsych/jsPsych#1619] [Q&A]
Hi, I am having trouble creating a looping timeline. I want a 'chunk' of trials (called OBsAct_1) that I then push to my timeline. I want the chunk to repeat if on the last trial (end_screen) the participant presses key_repeat. But when I press a key the experiment stops and I get an error "jspsych not defined). I don't think it's a problem with 'compareKeys' because it also doesn't work if I use "if(.....response == key_repeat).

I am using the new jspsych 3.0.
```
   var ObsAct_1 = {  // looping node
     timeline: [obsact_1a, demo[0], obsact_1b, demo[1], obsact_1c, end_screen], // inst, demo, inst, demo, inst
     loop_function: function(data){
       if(jspsych.pluginAPI.compareKeys(data.values()[0].response, key_repeat)){
         return true;
       }else{
         return false;
       }
     },
   };
```
^^^ Here the "obsact_1a etc. are instruction type trials and demo are linear chunks of trials.

In a github article from 2015 I had found this code for looping timelines:
```
var looping_chunk = {
	chunk_type: 'while',
	timeline: [single_stim_block],
	continue_function: function(data){
```
(...etc.)
But this also didn't work so I guess that it might be outdated?

Thank you for any help!

**Comment by martinakavanova** [comment]
Sorry, so I figured out that it's a problem with the loop_function but with the compareKey plugin which is not working for me for some reason. If I use 
       if(data.last(1).values()[0].response == key_repeat.){
then it works!

**Reply by jodeleeuw** [reply]
I think the reason your `compareKeys` function wasn't working is that `jspsych` was all lower case instead of `jsPsych`.
